### ISSUE TITLE
az functionapp create returns error when trying to create a dotnet-isolated functionapp

### ISSUE THREAD

**Issue by dozer75** [Azure/azure-cli#22281]
**Related command**

`az functionapp create`

**Describe the bug**
When using the command below to create an isolated .NET 5 function app environment the `az functionapp create` returns immediately with the error message (line breaks added to make it easier to read the message)

```
Runtime dotnet-isolated not supported for os linux. Supported runtimes for os linux are: 
['dotnet', 'dotnet', 'node', 'node', 'node', 'python', 'python', 'python', 'java', 'java', 'powershell', 'powershell', 'custom']. 
Run 'az functionapp list-runtimes' for more details on supported runtimes.
```

_Please note the duplicate supported runtimes displayed as well._

**To Reproduce**

1. Have the Azure CLI version 2.36.0
2. Create required resources for the command to run (see the documentation from [Azure Functions dotnet worker](https://github.com/Azure/azure-functions-dotnet-worker#deploying-to-azure) repo)
3. Execute the following command and replace related `<>` values

```
az functionapp create `
  --resource-group <Resource group> `
  --name <Function name> `
  --storage-account "<Storage account>" `
  --consumption-plan-location westeurope `
  --functions-version 3 `
  --os-type Linux `
  --runtime dotnet-isolated `
  --runtime-version 5.0 `
  --app-insights-key <App-insights key> `
  --assign-identity
```

**Expected behavior**

The `dotnet-isolated` should be allowed as a runtime since it is supported by Azure Functions.

**Environment summary**

- Azure CLI version 2.36.0
  - ```
    az version
    {
      "azure-cli": "2.36.0",
      "azure-cli-core": "2.36.0",
      "azure-cli-telemetry": "1.0.6",
      "extensions": {
        "aks-preview": "0.5.48",
        "application-insights": "0.1.14",
        "log-analytics-solution": "0.1.1",
        "peering": "0.2.1"
      }
    }
     ```
  - installed using MSI
- Powershell 7.2.3

**Additional context**
<!--- Add any other context about the problem here. --->

- This identical command works in earlier versions of Azure CLI (2.31.0), but have for some reason stopped working and gives us the specified error.
- The error is the same if you try to execute it with `os-type` `Windows`
- The [Azure Function dotnet documentation](https://github.com/Azure/azure-functions-dotnet-worker#deploying-to-azure) states that this is correct command.
- For some reason it isn't possible to deploy isolated .NET 6 using 2.31.0 but that's another story (and generally a problem with .NET 6 and out of scope of this issue).

**Comment by yonzhan** [comment]
route to CXP team

**Comment by navba-MSFT** [comment]
@dozer75 Thanks for reaching out to us and sharing this feedback. I am looking in to this issue. I will update this thread once I have more details.

**Comment by navba-MSFT** [comment]
@dozer75 I have reached out to the Product owners with all the details. I will update this thread once I hear back from them.

**Comment by navba-MSFT** [comment]
@dozer75 I have got an update from the Product group that this is not a CLI issue. The CLI doesn't consider dotnet-isolated a valid function app runtime since the API has the isHidden parameter set to true for that runtime.

To avoid the confusion, we will be updating this documentation : https://github.com/Azure/azure-functions-dotnet-worker#create-the-azure-resources

Also you can use the AzCLI 2.34.0 version as a workaround for now. I was able to create a function app with dotnet-isolated runtime using this CLI version.  Hope this helps.

**Comment by dozer75** [comment]
@navba-MSFT Thanks for answer. Why is it hidden? What does that mean?

Additionally; unfortunately, there are other issues as well with Azure Function App and dotnet-isolated (see Azure/azure-functions-host#8367) so it is even with 2.34.0 the dotnet-isolated resource fails to be created most of the time which makes it impossible to work with. (I have tried to get in contact with the Azure Functions team on their Git repo due to other issues, but to no avail, they are really quiet there so I haven't got any answer)

**Comment by navba-MSFT** [comment]
@dozer75 Hidden as in, Its not exposed through the API. This is the same API that the portal uses – note that the portal does not have an option to create a dotnet-isolated function app. So as of now, the ONLY workaround is to use AzCLI 2.34.0. I have tested it and it does work. If you are using that version and still face the same issue that can be troubleshooted differently. I have also created the above PR to update below documentation so that it helps other customers. 

https://github.com/Azure/azure-functions-dotnet-worker#create-the-azure-resources

**Comment by navba-MSFT** [comment]
@dozer75 I wanted to do quick follow-up to check if you had a chance to look at my above comment and test the given workaround. Please let us know if you need any further assistance on this. Awaiting your reply.

**Comment by dozer75** [comment]
@navba-MSFT Sorry forgot to answer, while the command does work, Azure has big issues with this for some reason.. It is impossible to create a working Azure Function App with dotnet in isolated mode on Linux environment that is working for us in westeurope, northeurope or norwayeast. We hade to some degree success on Windows environment, but that to is also unstable. As mentioned earlier I have tried to take this issue in other channels as well but to no avail... So simply put, we gave up dotnet with isolated mode and have rewritten our code to use dotnet without isolated mode which works, it was quicker to do than to get aid to fix the issue on Azure.

**Comment by navba-MSFT** [comment]
@dozer75 We had reached out to the Product Owners in the background and they are aware of this issue. Hence just to unblock you the workaround was shared. We hear you and understand the concerns. Apologies for the inconvenience caused.

>We're discussing whether we truly want to make a change to our guidance as opposed to working with the CLI team to have the bug fixed. We'll provide an update here soon

Reference: https://github.com/Azure/azure-functions-dotnet-worker/pull/894

Also there's an issue tracking it https://github.com/Azure/azure-functions-dotnet-worker/issues/857 which has been created. There is no ETA on when it will be fixed, but it seems the Stacks API fix is blocked on some other host issue.
